crates/sign/src/lib.rs

```diff
@@ -56,6 +56,18 @@ fn verify_message(msg: &Message, signature: &Signature) -> Result<(), secp256k1:
     secp.verify_ecdsa(msg, &sig, &pk)
 }
 
+/// Verify the given [`PublicKey`] against the message & signature.
+pub fn verify_signature(
+    msg: &Message,
+    sig: &Signature,
+    pk: &PublicKey,
+) -> Result<(), secp256k1::Error> {
+    let secp = Secp256k1::verification_only();
+    let compact_sig: CompactSignature =
+        CompactSignature::from_compact(&sig.0).expect("bytes slice is too short");
```

Comment:
```suggestion
    let compact_sig: CompactSignature = CompactSignature::from_compact(&sig.0)?;
```
should we forward the error?